torch/csrc/jit/pybind_utils.h

```diff
@@ -170,6 +170,14 @@ inline InferredType tryToInferContainerType(py::handle input) {
         return type_match.reason();
       }
     }
+    if (PyObject_HasAttrString(input.ptr(), "_fields")) {
+      // If this is a NamedTuple type, fetch the names correspondingly.
+      auto qualifiedName = c10::QualifiedName(py::cast<std::string>(
```

Comment:
unfortunately we could not call `get_named_tuple_property` function here because it will try to resolve the python type annotation again where it does not exist in tracing (tracing does not have compilation unit to hold registered types).

Reply:
This code is going to create a new NamedTuple type for every input value, even if it has the same type as another named tuple object in the system. This will cause future bugs. E.g. a scripted network that contains a traced sub-network that contains named tuples types. The traced sub-network will not be able to ever invoke the trace because the named tuple will have been resolved to a different type. Tracing should re-use the same infrastructure that named tuples and classes use in scripting. It can use the python_compilation_unit that all other python code uses.